packages/auto-approve/src/auto-approve.ts

```diff
@@ -144,82 +132,100 @@ export function handler(app: Probot) {
         prNumber
       );
 
-      // If there's no config, check to see if the PR is creating the auto-approve.yml file
-      if (!config) {
-        const prConfig = await getBlobFromPRFiles(
+      // Check to see if the config is being modified in the PR, before we check
+      // if it exists in the repo. If it's being modified, we want to submit
+      // a check
+      const prConfig = await getBlobFromPRFiles(
+        context.octokit,
+        owner,
+        repo,
+        PRFiles,
+        `.github/${CONFIGURATION_FILE_PATH}`
+      );
+
+      if (prConfig) {
+        // Attempt to get the CODEOWNERS file if it exists
+        const codeOwnersFile = await getBlobFromPRFiles(
           context.octokit,
           owner,
           repo,
           PRFiles,
-          `.github/${CONFIGURATION_FILE_PATH}`
+          '.github/CODEOWNERS'
         );
 
-        if (prConfig) {
-          // Attempt to get the CODEOWNERS file if it exists
-          const codeOwnersFile = await getBlobFromPRFiles(
-            context.octokit,
-            owner,
-            repo,
-            PRFiles,
-            '.github/CODEOWNERS'
-          );
-
-          // Decide whether to add a passing or failing status checks
-          await validateConfig(
+        // Decide whether to add a passing or failing status checks
+        await validateConfig(
+          owner,
+          repo,
+          prConfig,
+          codeOwnersFile,
+          context.octokit,
+          context.payload.pull_request.head.sha
+        );
+      } else {
+        let config: Configuration | null;
+        // Get auto-approve.yml file if it exists
+        // Reading the config requires access to code permissions, which are not
+        // always available for private repositories.
+        try {
+          config = await context.config<Configuration>(CONFIGURATION_FILE_PATH);
+        } catch (err) {
+          err.message = `Error reading configuration: ${err.message}`;
+          logger.error(err);
+          config = null;
+        }
+        // If there is a config, first confirm that it matches the guidelines
+        // Then, check to see whether the incoming PR matches the config
+        if (config) {
+          // If config is not valid, this function will submit a failing check, and will not merge the PR
+          const isConfigValid = await validateConfig(
```

Comment:
Great question! This actually has to do with my response to @chingor13's review. `validateConfig` checks whether the config is valid, creates a check reflecting whether the config is valid, and returns a boolean whether the config is valid. In the first code block (starting at line 146), we're checking whether a pull request is changing the auto-approve.yml config. If it does, then _all_ we need to do is create that check, and that's it -- we explicitly **don't** want to do anything else with that pull request, like approve or merge it. However, if the PR is changing other files other than `auto-approve.yml`, then we want to check if the configuration is valid _and_ that the PR matches the configuration.  This extra check that the configuration is valid comes from @tmatsuo's suggestion to make sure the config is valid before automerging pull requests that match the config, since someone with admin rights or someone who just ignores the failing auto-approve check might merge an invalid config. In that case, we might be automerging pull requests that match a config that we don't allow (e.g., different authors, etc.).

From @chingor13 and your comments, it sounds like this flow might be unclear. Are there any comments/code improvements I could add to make this flow a bit clearer?

Reply:
Sounds good.

Please add a a one or two-sentence comment to line 157 explaining why the return value is not examined.